etc/init.d/dscpclassify

```diff
@@ -45,173 +127,229 @@ iifzone() {
 		return 1
 	}
+	if [ -n "$ipset" ] && [ -n "$ipv4$ipv6" ]; then
+		log warning "Rules should not mix IP addresses and sets in the src_ip option"
+		return 1
+	fi
 	if [ -n "$ipv4" ] && [ "$2" = "ipv6" ]; then
-		logger -t dscpclassify "Rules cannot combine an ipv4 src_ip with the ipv6 family"
+		log warning "Rules cannot combine an ipv4 src_ip with the 'ipv6' family option"
 		return 1
 	fi
 	if [ -n "$ipv6" ] && [ "$2" = "ipv4" ]; then
-		logger -t dscpclassify "Rules cannot combine an ipv6 src_ip with the ipv4 family"
+		log warning "Rules cannot combine an ipv6 src_ip with the 'ipv4' family option"
 		return 1
 	fi
 
-	[ -n "$ipv4" ] && saddr="ip saddr { $(mklist "$ipv4") }"
-	[ -n "$ipv6" ] && saddr6="ip6 saddr { $(mklist "$ipv6") }"
-	return 0
+	[ -n "$ipv4" ] && daddr="ip saddr { $(mklist "$ipv4") }"
+	[ -n "$ipv6" ] && daddr6="ip6 saddr { $(mklist "$ipv6") }"
+
+	[ -n "$ipset" ] || return 0
+	case $2 in
+	ipv4) daddr="ip saddr $ipset" ;;
+	ipv6) daddr6="ip6 saddr $ipset" ;;
+	esac
 }
 
 device() {
 	[ -n "$1$2" ] || return 0
+
 	if [ -z "$1" ] || [ -z "$2" ]; then
-		logger -t dscpclassify "Rules must use the device and direction options in conjunction"
+		log warning "Rules must use the device and direction options in conjunction"
 		return 1
 	fi
-	[ "$2" = "in" ] && {
-		iifname "$1"
-		return 0
-	}
-	[ "$2" = "out" ] && {
-		oifname "$1"
-		return 0
-	}
-	logger -t dscpclassify "The direction rule option must be either 'in' or 'out'"
-	return 1
 
+	case $2 in
+	in) iifname "$1" ;;
+	out) oifname "$1" ;;
+	*)
+		log warning "The direction rule option must contain either 'in' or 'out'"
+		return 1
+		;;
+	esac
 }
 
 verdict() {
 	local class
 
 	[ -n "$1" ] || {
-		logger -t dscpclassify "Rule missing a dscp class"
+		log warning "Rule is missing the DSCP class option"
+		return 1
+	}
+	class="$(check_class "$1")" || {
+		log warning "Rule contains an invalid DSCP class"
 		return 1
 	}
 
-	class="$(echo "$1" | tr 'A-Z' 'a-z')"
-	[ "$class" = "be" ] && class="cs0"
+	verdict="goto ct_set_$class"
+}
 
-	case "$class" in
-	"cs0" | "le" | "cs1" | "af11" | "af12" | "af13" | "cs2" | "af21" | "af22" | "af23" | "cs3" | "af31" | "af32" | "af33" | "cs4" | "af41" | "af42" | "af43" | "cs5" | "va" | "ef" | "cs6" | "cs7")
-		verdict="goto ct_set_$class"
-		return 0
+comment() {
+	[ -n "$1" ] || return 0
+	comment="comment \"$1\""
+}
+
+pre_include() {
+	echo "$1" >>"/tmp/etc/dscpclassify-pre.include"
+}
+
+post_include() {
+	echo "$1" >>"/tmp/etc/dscpclassify-post.include"
+}
+
+create_user_set() {
+	local constant element enabled interval family flags name timeout type
+
+	config_get_bool enabled "$1" enabled
+	[ "$enabled" = 0 ] && return 0
+
+	config_get_bool constant "$1" constant
+	config_get element "$1" element
+	config_get_bool interval "$1" interval
+	config_get family "$1" family
+	config_get name "$1" name
+	config_get timeout "$1" timeout
+	config_get type "$1" type
+
+	if [ -z "$famiy$type" ] || [ -n "$family" ] && [ -n "$type" ]; then
```

Comment:
typo - should be 'family'

Reply:
Fixed in [6c4cef4](https://github.com/jeverley/dscpclassify/pull/5/commits/6c4cef41e12a1fee9f89b4eb1d0c5b0df99ca937)